include/core/shape.h

```diff
@@ -757,6 +839,190 @@ class CompositeShape : public Shape<dim>
     operations;
 };
 
+template <int dim>
+class OpenCascadeShape : public Shape<dim>
+{
+public:
+  /**
+   * @brief Constructor for an OpenCascade shape
+   * @param file_name The name of the file describing the shape
+   * @param position The shape center
+   * @param orientation The shape orientation
+   */
+  OpenCascadeShape<dim>(std::string         file_name,
+                        const Point<dim> &  position,
+                        const Tensor<1, 3> &orientation)
+    : Shape<dim>(0.1, position, orientation)
+  {
+    // First, we read the shape file name
+    local_file_name = file_name;
+#ifdef DEAL_II_WITH_OPENCASCADE
+    // Checks the file name extension to identify which type of OpenCascade
+    // shape we are working with.
+    std::vector<std::string> file_name_and_extension(
+      Utilities::split_string_list(local_file_name, "."));
+
+    // Load the shape with the appropriate tool.
+    if (file_name_and_extension[1] == "step" ||
+        file_name_and_extension[1] == "stp")
+      {
+        shape = OpenCASCADE::read_STEP(local_file_name);
+        this->additional_info_on_shape = "step";
+      }
+    else if (file_name_and_extension[1] == "iges" ||
+             file_name_and_extension[1] == "igs")
+      {
+        shape = OpenCASCADE::read_IGES(local_file_name);
+        this->additional_info_on_shape = "iges";
+      }
+    else if (file_name_and_extension[1] == "stl")
+      {
+        shape                          = OpenCASCADE::read_STL(local_file_name);
+        this->additional_info_on_shape = "stl";
+      }
+    else
+      {
+        throw "Wrong file extension for an opencascade shape. The possible file type are: step, stp, iges, igs, stl.";
+      }
+
+    // used this local variable as the shape tolerance in the calculations.
+    shape_tol = OpenCASCADE::get_shape_tolerance(shape);
+
+    // Initialize some variables and the OpenCascade distance tool.
+    OpenCASCADE::extract_compound_shapes(
+      shape, compounds, compsolids, solids, shells, wires);
+    vertex_position = OpenCASCADE::point(Point<dim>());
+    vertex          = BRepBuilderAPI_MakeVertex(vertex_position);
+    distancetool    = BRepExtrema_DistShapeShape(shape, vertex);
+
+    // Check if the shape has a shell. If it has a shell, we initialize a
+    // distance tool with just the shell.
+    if (shells.size() > 0)
+      {
+        if (shells.size() > 1)
+          {
+            std::cout
```

Comment:
Avoid using std::cout, what happens if this is run in parallel? Recreate a pcout locally using the mpi_communicator if need be or write only from proc 0.